### PR TITLE
Save SMTP server last response not only after message sending but aft…

### PR DESCRIPTION
**Pull request by disaykin**
…er login too

### CODE REVIEW

**Comment by dinhviethoa** [comment]
Could you clear our the last response in connectIfNeeded()?

**Comment by disaykin** [comment]
For example, smtp.wp.pl server sometimes respond this way:

`dmitrijisajkin$ telnet smtp.wp.pl 465
Trying 212.77.101.1...
Connected to smtp.wp.pl.
Escape character is '^]'.
550 [IP:62.210.105.116] polaczenia z Twojego adresu IP sa zablokowane administracyjnie / mail from your IP address is administratively disabled
Connection closed by foreign host.`

For now, this situation treats by libetpan as SSL handshake failure. But in the future, somebody may improve this code, and libetpan will be return "550 [IP:62.210.105.116] polaczenia z Twojego adresu IP sa zablokowane administracyjnie / mail from your IP address is administratively disabled" as response string. This response may be shown to user as hint about problem resolution.

**Comment by dinhviethoa** [comment]
I meant, when starting a new command, we need to clean up the old response from the previous command.
I think the best location would be in connectIfNeeded().

**Comment by disaykin** [comment]
Done.

File: src/core/smtp/MCSMTPSession.cpp [hunk 1]
```diff
@@ -717,17 +745,16 @@ void SMTPSession::internalSendMessage(Address * from, Array * recipients, Data *
         mailsmtp_quit(mSmtp);
     }
     esmtp_address_list_free(address_list);
+    saveLastResponse();
 
     String * response;
```
Comment (dinhviethoa) [inline src/core/smtp/MCSMTPSession.cpp]: remove these 8 lines altogether. They don't seem to be useful.

**Comment by disaykin** [comment]
Done

File: src/core/smtp/MCSMTPSession.cpp [hunk 2]
```diff
@@ -717,25 +745,15 @@ void SMTPSession::internalSendMessage(Address * from, Array * recipients, Data *
...
+    saveLastResponse();
 
     if ((r == MAILSMTP_ERROR_STREAM) || (r == MAILSMTP_ERROR_CONNECTION_REFUSED)) {
         * pError = ErrorConnection;
         mShouldDisconnect = true;
         goto err;
     }
     else if (r == MAILSMTP_ERROR_EXCEED_STORAGE_ALLOCATION) {
-        if ((response != NULL) && (response->locationOfString(MCSTR("5.7.0")) != -1)) {
+        if ((mLastSMTPResponse != NULL) && (mLastSMTPResponse->locationOfString(MCSTR("5.7.0")) != -1)) {
```
Comment (dinhviethoa) [inline src/core/smtp/MCSMTPSession.cpp]: ha, my bad, it was still used here! Anyway, it's good either way. I'm merging the PR.

**Comment by dinhviethoa** [comment]
Thanks!

**Pull request merged**